x-pack/plugins/lens/public/xy_visualization/xy_config_panel.tsx

```diff
@@ -118,14 +121,114 @@ export function LayerContextMenu(props: VisualizationLayerWidgetProps<State>) {
 }
 
 export function XyToolbar(props: VisualizationToolbarProps<State>) {
+  const axes = [
+    {
+      id: 'x',
+      label: 'X-axis',
+    },
+    {
+      id: 'y',
+      label: 'Y-axis',
+    },
+  ];
+
+  const { frame, state, setState } = props;
+
   const [open, setOpen] = useState(false);
-  const hasNonBarSeries = props.state?.layers.some(
+  const hasNonBarSeries = state?.layers.some(
     (layer) => layer.seriesType === 'line' || layer.seriesType === 'area'
   );
+
+  const [xaxistitle, setXaxistitle] = useState(state?.xTitle);
+  const [yaxistitle, setYaxistitle] = useState(state?.yTitle);
+
+  const xyTitles = useCallback(() => {
+    const defaults = {
+      xTitle: state?.xTitle,
+      yTitle: state?.yTitle,
+    };
+    const layer = state?.layers[0];
+    if (!layer || !layer.accessors.length) {
+      return defaults;
+    }
+    const datasource = frame.datasourceLayers[layer.layerId];
+    if (!datasource) {
+      return defaults;
+    }
+    const x = layer.xAccessor ? datasource.getOperationForColumnId(layer.xAccessor) : null;
+    const y = layer.accessors[0] ? datasource.getOperationForColumnId(layer.accessors[0]) : null;
+
+    return {
+      xTitle: defaults.xTitle || x?.label,
+      yTitle: defaults.yTitle || y?.label,
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [open]);
```

Comment:
Is there a reason you are not specifying exhaustive deps here?

Reply:
We want is this to be called only when open changes state because otherwise when the user has the settings panel open, and clears the input then the default label will be filled in. But we want the default value to be filled in only if the user closes and reopens the panel

Reply:
Ah, OK, that makes sense, thanks. In that case the local state is fine as well as it follow from that. Could you add a comment describing that?

Reply:
Of course!